scripts/cluster-upgrade.sh

```diff
@@ -469,32 +469,12 @@ check_cluster_status() {
 
     # get a list of firing critical alerts in core namespaces (these would alert SREP in PD)
     # NOTE exclude ClusterUpgradingSRE, DNSErrors05MinSRE and MetricsClientSendFailingSRE
-    CA=$(curl -G -s -k -H "Authorization: Bearer $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring sa get-token prometheus-k8s)" --data-urlencode "query=ALERTS{alertstate=\"firing\",severity=\"critical\",namespace=~\"^openshift.*|^kube.*|^default$\",namespace!=\"openshift-customer-monitoring\",alertname!=\"ClusterUpgradingSRE\",alertname!=\"DNSErrors05MinSRE\",alertname!=\"MetricsClientSendFailingSRE\"}" "https://$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring get routes prometheus-k8s -o json | jq -r .spec.host)/api/v1/query" | jq -r '.data.result[].metric.alertname' | tr '\n' ',' )    
+    CA=$(curl -G -s -k -H "Authorization: Bearer $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring sa get-token prometheus-k8s)" --data-urlencode "query=ALERTS{alertstate=\"firing\",severity=\"critical\",namespace=~\"^openshift.*|^kube.*|^default$\",namespace!=\"openshift-customer-monitoring\",alertname!=\"ClusterUpgradingSRE\",alertname!=\"DNSErrors05MinSRE\",alertname!=\"MetricsClientSendFailingSRE\"}" "https://$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring get routes prometheus-k8s -o json | jq -r .spec.host)/api/v1/query" | jq -r '.data.result[].metric.alertname' | tr '\n' ',' )
 
     if [ "$CA" != "" ];
     then
         log $OCM_NAME $LOG_NAME "WARNING: Critical Alerts = $CA"
         CLUSTER_STATUS=0
-
-        # degraded operators
-        DCO=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get clusteroperator --no-headers | awk '{print $1 "," $5}' | grep -v ",False" | awk -F, '{print $1 ","}' | xargs)
-        
-        if [ "$DCO" != "" ];
-        then
-            log $OCM_NAME $LOG_NAME "INFO: Degraded Operators = $DCO"
-
-            for CO in $DCO;
-            do
-                # get clusteroperator error if we can
-                CO_MESSAGE=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get clusteroperator $CO -o json 2>/dev/null | jq -r '.status.extension.lastSyncError')
-
-                if [ "$CO_MESSAGE" != "" ];
-                then
-                    log $OCM_NAME $LOG_NAME "INFO: Operator $CO lastSyncError = $CO_MESSAGE"
-                fi
-            done
-
-        fi
```

Comment:
Looking at this again, this will not change the upgrade script.  This is going to only be triggered when there's a critical alert already active.  This is only in here to report any degraded operators when there is a critical alert to help provide context to whoever is doing the upgrade.  I'm not in favor of removing this since it's not blocking OSD upgrades.

Reply:
Agreed - this output was actually the main way I was able to tell that something was not right with some of the upgrades, when an operator reported as degraded for an extended period of time. During the mid-upgrade process the cluster_status() call is for informational purposes only, it doesn't affect progression.